addons/arsenal/Cfg3DEN.hpp

```diff
@@ -0,0 +1,188 @@
+                    text = SYMBOL_ITEM_VIRTUAL;
+                };
+                class SearchButton: ctrlButtonPicture {
+                    idc = IDC_ATTRIBUTE_SEARCH_BUTTON;
+                    onButtonClick = QUOTE(((ctrlParentControlsGroup (_this select 0)) controlsGroupCtrl IDC_ATTRIBUTE_SEARCHBAR) ctrlSetText ''; [ctrlParentControlsGroup (_this select 0)] call FUNC(attributeAddItems));
+                    text = "\a3\Ui_f\data\GUI\RscCommon\RscButtonSearch\search_start_ca.paa";
+                    x = QUOTE(5 * ATTRIBUTE_W);
+                    y = QUOTE(101.83 * ATTRIBUTE_H);
+                    w = QUOTE(5 * ATTRIBUTE_W);
+                    h = QUOTE(5 * ATTRIBUTE_H);
+                    colorBackground[] = {0, 0, 0, 0.5};
+                };
+                class SearchBar: ctrlEdit {
+                    idc = IDC_ATTRIBUTE_SEARCHBAR;
+                    onKeyUp = QUOTE([ctrlParentControlsGroup (_this select 0)] call FUNC(attributeAddItems));
+                    x = QUOTE(11 * ATTRIBUTE_W);
+                    y = QUOTE(101.83 * ATTRIBUTE_H);
+                    w = QUOTE(55 * ATTRIBUTE_W);
+                    h = QUOTE(5 * ATTRIBUTE_H);
+                };
+                class ClearButton: ctrlButton {
+                    idc = IDC_ATTRIBUTE_CLEAR_BUTTON;
+                    onButtonClick = QUOTE([ctrlParentControlsGroup (_this select 0)] call FUNC(attributeClear));
+                    text = "$STR_disp_arcmap_clear";
+                    x = QUOTE(105 * ATTRIBUTE_W);
+                    y = QUOTE(101.83 * ATTRIBUTE_H);
+                    w = QUOTE(25 * ATTRIBUTE_W);
+                    h = QUOTE(5 * ATTRIBUTE_H);
+                    colorBackground[] = {0, 0, 0, 0.6};
+                };
+                class ExportButton: ClearButton {
+                    idc = IDC_ATTRIBUTE_EXPORT_BUTTON;
+                    onButtonClick = QUOTE(copyToClipboard str ((uiNamespace getVariable [ARR_2(QQGVAR(attributeValue),[ARR_2([],0)])]) select 0));
+                    text = CSTRING(buttonExportText);
+                    tooltip = CSTRING(AttributeExport_Tooltip);
+                    x = QUOTE(79 * ATTRIBUTE_W);
+                };
+            };
+        };
+    };
+    class Object {
+        class AttributeCategories {
+            class ADDON {
+                displayName = CSTRING(Mission);
+                collapsed = 1;
+                class Attributes {
+                    class ADDON {
+                        property = QGVAR(attribute);
+                        control = QGVAR(attribute);
+                        displayName = CSTRING(Mission);
+                        tooltip = "";
+                        expression = QUOTE(if (!is3DEN) then {[ARR_2(_this,+_value)] call FUNC(attributeInit)});
+                        defaultValue = "[[], 0]";
+                        condition = "1";
```

Comment:
You can add it to soldiers, which works for accessing from other units interaction, but you can't access yourself.
Should we limit to just containers `objectHasInventoryCargo`
or just not soldiers `1 - objectControllable`
or add a self interaction action when added to CaManBase
or maybe it's not really a problem?

Reply:
I think `1 - objectControllable` would be the best option.